docs/templates/process-logging.md

```diff
@@ -0,0 +1,292 @@
+# Workspace Process Logging
+
+The workspace process logging feature allows you to log all system-level
+processes executing in the workspace.
+
+> **Note:** This feature is only available on Linux in Kubernetes. There are
+> additional requirements outlined further in this document.
+>
+> This is an Enterprise feature. To learn more about Coder Enterprise, please
+> contact sales.
+
+Workspace process logging adds a sidecar container to workspace pods that will
+log all processes started in the workspace container (e.g., commands executed in
+the terminal or processes created in the background by other processes).  You
+can view the output from the sidecar or send it to a monitoring stack, such as
+CloudWatch, for further analysis or long-term storage.
+
+Please note that these logs are not recorded or captured by the Coder
+organization in any way, shape, or form.
+
+## How this works
+
+Coder uses [eBPF](https://ebpf.io/) (which we chose for its minimal performance
+impact) to perform in-kernel logging and filtering of all exec system calls
+originating from the workspace container.
+
+The core of this feature is also open source and can be found in the
+[exectrace](https://github.com/coder/exectrace) repo on GitHub repo. The
+enterprise component (in the `enterprise/` directory of the repo) is responsible
+for starting the eBPF program with the correct filtering options for the
+specific workspace.
+
+## Requirements
+
+The host machine must be running a Linux kernel >= 5.8 with the kernel config
+`CONFIG_DEBUG_INFO_BTF=y` enabled.
+
+To check your kernel version, run:
+
+```shell
+$ uname -r
+```
+
+To validate the required kernel config is enabled, run either of the following
+commands on your nodes directly (_not_ from a workspace terminal):
+
+```shell
+$ cat /proc/config.gz | gunzip | grep CONFIG_DEBUG_INFO_BTF
+```
+
+```shell
+$ cat "/boot/config-$(uname -r)" | grep CONFIG_DEBUG_INFO_BTF
+```
+
+If these requirements are not met, workspaces will fail to start for security
+reasons.
+
+Your template must be a Kubernetes template. Workspace process logging is not
+compatible with the `sysbox-runc` runtime due to technical limitations, but it
+is compatible with our `envbox` template family.
+
+## Example templates
+
+We provide working example templates for Kubernetes, and Kubernetes with
+`envbox` (for [Docker support in workspaces](./docker-in-workspaces.md)). You
+can view these templates in the
+[exectrace repo](https://github.com/coder/exectrace/tree/main/enterprise/templates).
```

Comment:
Any ideas on how to avoid drift between these example templates and the code snippets in our docs? Perhaps we add a comment to both places informing users to update the other place?